core/utils/histogram.h

```diff
@@ -147,24 +196,28 @@ class Histogram {
   // Resets all counters, and the count of such counters.
   // Note that the number of buckets remains unchanged.
   void Reset() {
-    count_ = 0;
+    count_.store(0, std::memory_order_relaxed);
     std::fill(buckets_.begin(), buckets_.end(), T());
   }
 
   // Resize the histogram.  Note that this resets it (i.e., this
   // does not attempt to redistribute existing counts).
   void Resize(size_t num_buckets, T bucket_width) {
-    buckets_.clear();
-    buckets_.resize(num_buckets + 1);
+    buckets_ = std::move(std::vector<std::atomic<uint64_t>>(num_buckets + 1));
     bucket_width_ = bucket_width;
-    count_ = 0;
+    count_.store(0, std::memory_order_relaxed);
   }
 
  private:
+  void nonatomic_inc(std::atomic<uint64_t> *p) {
```

Comment:
minor nit: if we ditch the counter it would be better to embed the body of these functions in `(Atomic)Insert`.